include/crow/utility.h

```diff
@@ -897,5 +896,48 @@ namespace crow
 
             return v.substr(begin, end - begin);
         }
+
+        class call_error : public std::runtime_error
+        {
+        public:
+            call_error() : std::runtime_error{"call error"}
+            {
+            }
+        };
+
+        template <typename ReturnType, typename... Args>
+        class lambda_wrapper
+        {
+        public:
+            lambda_wrapper() = default;
+
+            lambda_wrapper(std::function<ReturnType(Args...)> func) : valid_(true), func_(func)
+            {
+            }
+
+            void clear() { valid_ = false; }
+
+            operator bool() const { return valid_; }
+
+            lambda_wrapper<ReturnType, Args...>& operator=(std::function<ReturnType(Args...)> func)
+            {
+                valid_ = true;
+                func_ = func;
+                return *this;
+            }
+
+            ReturnType operator()(Args... args) const
+            {
+                if (valid_)
+                {
+                    return func_(args...);
+                }
+                throw call_error();
+            }
```

Comment:
I don't really understand what lambda_wrapper does 🙂 After all, std just throws as well 

https://en.cppreference.com/w/cpp/utility/functional/function/operator()

> Exceptions
Throws [std::bad_function_call](https://en.cppreference.com/w/cpp/utility/functional/bad_function_call) if *this does not store a callable function target, i.e. !*this == true.